src/pages/SignupPage.jsx

```diff
@@ -0,0 +1,135 @@
+import React, { useState, useContext } from "react";
+import { Link, useNavigate } from "react-router-dom";
+import { firebaseAuth, firestore } from "../service/firebase";
+import { createUserWithEmailAndPassword } from "firebase/auth";
+import { setDoc, doc } from "firebase/firestore";
+import Button from "../Components/Button/Button";
+import styles from "./AuthForm.module.css";
+import { ThemeContext } from "../Contexts/ThemeContext";
+import { authFailMessageMap } from "../static/maps";
+import NoticeModal from "../Components/Modal/NoticeModal";
+import Spinner from "../Components/Loaders/Spinner";
+
+const SignupPage = () => {
+  const { theme } = useContext(ThemeContext);
+  const navigate = useNavigate();
+  const [isLoading, setIsLoading] = useState(false);
+  const [showFail, setShowFail] = useState(false);
+  const [failMessage, setFailMessage] = useState("");
+  const [input, setInput] = useState({
+    email: "",
+    displayName: "",
+    password: "",
+  });
+  const labelClass = `${styles.label} convex-${theme}-md`;
+  const inputClass = `${styles.input} concave-${theme}-md`;
+
+  const handleSignup = async (e) => {
+    e.preventDefault();
+    if (input.displayName === "") {
+      setShowFail(true);
+      setFailMessage("Enter a nickname!");
+      return;
+    }
+    setIsLoading(true);
+    await createUserWithEmailAndPassword(
+      firebaseAuth,
+      input.email,
+      input.password
+    )
+      .then(async (userCredential) => {
+        const user = userCredential.user;
+        await setDoc(doc(firestore, "users", user.uid), {
+          uid: user.uid,
+          ...input,
+        });
+        setIsLoading(false);
+        navigate("/");
+      })
+      .catch((error) => {
+        setIsLoading(false);
+        setShowFail(true);
+        setFailMessage(authFailMessageMap[error.code]);
+      });
```

Comment:
특별한 상황이 아니라면 async/awiat과 then/catch를 혼용하지 않는게 가독성이 더 좋습니다.
```suggestion
    try {
      cosnt userCredential = await createUserWithEmailAndPassword(
        firebaseAuth,
        input.email,
        input.password
      )
      const user = userCredential.user;
      await setDoc(doc(firestore, "users", user.uid), {
        uid: user.uid,
        ...input,
      });
      setIsLoading(false);
      navigate("/");
   } catch {
      setIsLoading(false);
        setShowFail(true);
        setFailMessage(authFailMessageMap[error.code]);
   }
```

Reply:
너무좋습니다 바로 적용해보겠습니다!